<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>R functions to use in the UK Biobank Research Analysis Platform (RAP) • ukbrapR</title>
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="R functions to use in the UK Biobank Research Analysis Platform (RAP)">
<meta name="description" content='R functions to use in the UK Biobank Research Analysis Platform (RAP). The aim is to make working in the RAP quicker, easier, and more reproducible. The package is designed to work in a "normal" cluster using RStudio, and raw UK Biobank data from the table-exporter (rather than on a Spark cluster).'>
<meta property="og:description" content='R functions to use in the UK Biobank Research Analysis Platform (RAP). The aim is to make working in the RAP quicker, easier, and more reproducible. The package is designed to work in a "normal" cluster using RStudio, and raw UK Biobank data from the table-exporter (rather than on a Spark cluster).'>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">ukbrapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/ascertain_diagnoses.html">Ascertain diagnoses</a></li>
    <li><a class="dropdown-item" href="articles/label_fields.html">Label fields</a></li>
    <li><a class="dropdown-item" href="articles/spark_functions.html">Spark functions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcpilling/ukbrapR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="ukbrapr-">ukbrapR <a href="https://lcpilling.github.io/ukbrapR/"><img src="reference/figures/ukbrapR.png" align="right" width="150"></a>
<a class="anchor" aria-label="anchor" href="#ukbrapr-"></a>
</h1></div>
<!-- badges: start -->

<p>ukbrapR (phonetically: ‘U-K-B-wrapper’) is an R package for working in the UK Biobank Research Analysis Platform (RAP). The aim is to make it quicker, easier, and more reproducible.</p>
<blockquote>
<p>Since <code>v0.2.0</code> ukbrapR works best on a “normal” cluster using RStudio and raw data from the table-exporter. Old Spark functions are still available but are not updated.</p>
</blockquote>
<p><sub>Wrapped server icon by DALL-E</sub></p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>In the DNAnexus Tools menu launch Posit Workbench and start an RStudio environment.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install current version</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lcpilling/ukbrapR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h2>
<p>There are three main groups of functions:</p>
<ul>
<li>
<span class="emoji" data-emoji="dna">🧬</span> <a href="#genetic-variants">Genetics</a>: extract genotypes from Bulk data, create polygenic score</li>
<li>
<span class="emoji" data-emoji="clipboard">📋</span> <a href="#ascertain-diagnoses">Diagnoses</a>: ascertain from health records and self-reported illness data, determine date first diagnosed</li>
<li>
<span class="emoji" data-emoji="hammer_and_wrench">🛠️</span> <a href="#utility-functions">Utilities</a>: Check field names, label data fields, upload/download files from RAP, and pull phenotypes from Spark</li>
</ul>
</div>
<div class="section level2">
<h2 id="genetic-variants">Genetic variants<a class="anchor" aria-label="anchor" href="#genetic-variants"></a>
</h2>
<p>Bulk imputed genotypes and variant calls from Whole Genome Sequencing are available and can be easily accessed in an RStudio instance.</p>
<div class="section level3">
<h3 id="extract-variants">Extract variants<a class="anchor" aria-label="anchor" href="#extract-variants"></a>
</h3>
<p><code><a href="reference/extract_variants.html">extract_variants()</a></code> by default uses <a href="https://enkre.net/cgi-bin/code/bgen" class="external-link">bgenix</a> and <a href="https://www.cog-genomics.org/plink/" class="external-link">plink</a> to subset the <a href="https://biobank.ctsu.ox.ac.uk/crystal/field.cgi?id=22828" class="external-link">imputed BGEN files</a> and read it quickly and easily into R.</p>
<p>The only required input is a data frame (or path to file) containing “rsid” and “chr” variables. See function documentation for further details/options (including the available <code><a href="reference/make_imputed_bed.html">make_imputed_bed()</a></code> and <code><a href="reference/load_bed.html">load_bed()</a></code> internal functions).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">varlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>rsid<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs1800562"</span>,<span class="st">"rs429358"</span><span class="op">)</span>, chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">19</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">imputed_genotypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_variants.html">extract_variants</a></span><span class="op">(</span><span class="va">varlist</span><span class="op">)</span></span>
<span><span class="co">#&gt; ~10 seconds</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">imputed_genotypes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 487409      3</span></span></code></pre></div>
<p>By setting option <code>source="dragen"</code> the function will instead subset the <a href="https://biobank.ctsu.ox.ac.uk/crystal/field.cgi?id=24309" class="external-link">DRAGEN WGS BGEN files</a>. This requires “pos” in the input data frame (build 38).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">varlist_b38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>rsid<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs1800562"</span>,<span class="st">"rs429358"</span><span class="op">)</span>, chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">19</span><span class="op">)</span>, pos<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">26092913</span>,<span class="fl">44908684</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dragen_genotypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_variants.html">extract_variants</a></span><span class="op">(</span><span class="va">varlist_b38</span>, source<span class="op">=</span><span class="st">"dragen"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ~15 seconds </span></span></code></pre></div>
<p>The highlight of developing this feature was naming the internal function <code><a href="reference/make_dragen_bed.html">make_dragen_bed()</a></code> <span class="emoji" data-emoji="dragon">🐉</span> <span class="emoji" data-emoji="bed">🛏️</span></p>
<blockquote>
<p>This assumes your project has access to the WGS BGEN files released April 2025. If not, run <code>ukbrapR:::make_dragen_bed_from_pvcfs()</code> to use <a href="https://www.htslib.org/doc/tabix.html" class="external-link">tabix</a> and <a href="https://www.cog-genomics.org/plink/" class="external-link">plink</a> to subset the <a href="https://biobank.ctsu.ox.ac.uk/crystal/field.cgi?id=24310" class="external-link">DRAGEN WGS pVCF files</a>. Much slower.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="create-polygenic-score">Create polygenic score<a class="anchor" aria-label="anchor" href="#create-polygenic-score"></a>
</h3>
<p><code><a href="reference/create_pgs.html">create_pgs()</a></code> takes a data frame containing a list of variant associations with a trait and creates a weighted allele score using <a href="https://www.cog-genomics.org/plink/1.9/score" class="external-link">plink</a>. By default it uses the <a href="https://biobank.ctsu.ox.ac.uk/crystal/field.cgi?id=22828" class="external-link">imputed</a> genotypes.</p>
<p>The only required input is a data frame (or path to file) containing rsid, chr, pos, effect_allele, other_allele, beta. For DRAGEN pos is build 38.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># weights from GWAS of liver cirrhosis (Innes 2020 Gastroenterology doi:10.1053/j.gastro.2020.06.014)</span></span>
<span><span class="va">varlist_pgs</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"files"</span>, <span class="st">"pgs_liver_cirrhosis.txt"</span>, package<span class="op">=</span><span class="st">"ukbrapR"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">varlist_pgs</span><span class="op">)</span></span>
<span><span class="co">#&gt;   rsID         CHR       POS effect_allele other_allele effect_weight locus_name</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;         &lt;chr&gt;                &lt;dbl&gt; &lt;chr&gt;     </span></span>
<span><span class="co">#&gt; 1 rs2642438      1 220796686 A             G                   -0.177 MARC1     </span></span>
<span><span class="co">#&gt; 2 rs11925835     3  56831417 T             C                   -0.235 ARHGEF3   </span></span>
<span><span class="co">#&gt; 3 rs72613567     4  87310241 TA            T                   -0.166 HSD17B13  </span></span>
<span><span class="co">#&gt; 4 rs2954038      8 125495147 C             A                    0.16  TRIB1     </span></span>
<span><span class="co">#&gt; 5 rs11065384    12 120985482 T             C                    0.275 HNF1A     </span></span>
<span><span class="co">#&gt; 6 rs28929474    14  94378610 T             C                    0.561 SERPINA1  </span></span>
<span></span>
<span><span class="va">liver_pgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_pgs.html">create_pgs</a></span><span class="op">(</span></span>
<span>    in_file<span class="op">=</span><span class="va">varlist_pgs</span>,                     <span class="co"># can be a data frame or file path</span></span>
<span>    out_file<span class="op">=</span><span class="st">"liver_cirrhosis.imputed.pgs"</span>,  <span class="co"># {optional} prefix for created .bed and .tsv files</span></span>
<span>    pgs_name<span class="op">=</span><span class="st">"liver_cirrhosis_pgs"</span><span class="op">)</span>          <span class="co"># {optional} variable name</span></span>
<span><span class="co">#&gt; → Extracting 9 variants from 8 imputed files</span></span>
<span><span class="co">#&gt; ✔ PGS created! See file liver_cirrhosis.imputed.pgs.tsv</span></span>
<span><span class="co">#&gt; ~1 minute</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">liver_pgs</span><span class="op">$</span><span class="va">liver_cirrhosis_pgs</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt; 0.00000 0.06006 0.08200 0.08589 0.10722 0.26639</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="ascertain-diagnoses">Ascertain diagnoses<a class="anchor" aria-label="anchor" href="#ascertain-diagnoses"></a>
</h2>
<p>Diagnosis of conditions in UK Biobank participants come from multiple data sources. {ukbrapR} makes it fast and easy to ascertain diagnoses from multiple UK Biobank data sources in the DNAnexus Research Analysis Platform (RAP). Follow the below steps. See the website article for more details.</p>
<div class="section level3">
<h3 id="id_1-export-tables-of-raw-data">1. Export tables of raw data<a class="anchor" aria-label="anchor" href="#id_1-export-tables-of-raw-data"></a>
</h3>
<p>This only needs to happen once per project. Run <code><a href="reference/export_tables.html">export_tables()</a></code> to submit the <code>table-exporter</code> jobs to save the required files to the RAP persistent storage. ~10Gb of text files are created, costing ~£0.15 per month to store.</p>
</div>
<div class="section level3">
<h3 id="id_2-get-diagnoses-from-all-data-sources">2. Get diagnoses from all data sources<a class="anchor" aria-label="anchor" href="#id_2-get-diagnoses-from-all-data-sources"></a>
</h3>
<p>For a given set of diagnostic codes get the participant Electronic Medical Records (EMR) and self-reported illess data. Returns a list containing up to 6 data frames: the subset of the clinical files with matched codes.</p>
<p>Codes need to be provided as a data frame with two fields: <code>vocab_id</code> and <code>code</code>. Valid code vocabularies are:</p>
<ul>
<li>
<code>ICD10</code> (for searching HES diagnoses, cause of death, and cancer registry)</li>
<li>
<code>ICD9</code> (for searching older HES diagnosis data)</li>
<li>
<code>Read2</code> and <code>CTV3</code> (for GP clinical events)</li>
<li>
<code>OPCS3</code> and <code>OPCS4</code> (for HES operations)</li>
<li>
<code>ukb_cancer</code> and <code>ukb_noncancer</code> (for self-reported illness at UK Biobank assessments - all instances will be searched)</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example diagnostic codes for CKD </span></span>
<span><span class="va">codes_df_ckd</span> <span class="op">&lt;-</span> <span class="fu">ukbrapR</span><span class="fu">:::</span><span class="va">codes_df_ckd</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">codes_df_ckd</span><span class="op">)</span></span>
<span><span class="co">#&gt;   condition vocab_id  code</span></span>
<span><span class="co">#&gt; 1       ckd    ICD10 N18.3</span></span>
<span><span class="co">#&gt; 2       ckd    ICD10 N18.4</span></span>
<span><span class="co">#&gt; 3       ckd    ICD10 N18.5</span></span>
<span><span class="co">#&gt; ...</span></span>
<span></span>
<span><span class="co"># get diagnosis data - returns list of data frames (one per source)</span></span>
<span><span class="va">diagnosis_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_diagnoses.html">get_diagnoses</a></span><span class="op">(</span><span class="va">codes_df_ckd</span><span class="op">)</span> </span>
<span><span class="co">#&gt; 7 ICD10 codes, 40 Read2 codes, 37 CTV3 codes </span></span>
<span><span class="co">#&gt; ~2 minutes</span></span>
<span></span>
<span><span class="co"># N records for each source</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">diagnosis_list</span><span class="op">$</span><span class="va">gp_clinical</span><span class="op">)</span>  <span class="co">#  29,083</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">diagnosis_list</span><span class="op">$</span><span class="va">hesin_diag</span><span class="op">)</span>   <span class="co"># 206,390</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">diagnosis_list</span><span class="op">$</span><span class="va">death_cause</span><span class="op">)</span>  <span class="co">#   1,962</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_3-get-date-first-diagnosed">3. Get date first diagnosed<a class="anchor" aria-label="anchor" href="#id_3-get-date-first-diagnosed"></a>
</h3>
<p>Identify the date first diagnosed for each participant from any of datasets searched with <code><a href="reference/get_diagnoses.html">get_diagnoses()</a></code> (cause of death, HES diagnoses, GP clinical, cancer registry, HES operations, and self-reported illness fields).</p>
<p>Also included are:</p>
<ul>
<li>a <code>src</code> field indicating the source of the date of first diagnosis.</li>
<li>a <code>bin</code> field indicating the cases [1] and controls [0]. This relies on a small number of baseline fields also exported. The <code>df</code> field for the controls is the date of censoring (currently 30 October 2022).</li>
<li>a <code>bin_prev</code> field indicating whether the case was before the UK Biobank baseline assessment</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># for each participant, get Date First diagnosed with the condition</span></span>
<span><span class="co">#   {optional} add a prefix to the variable names with "prefix"</span></span>
<span><span class="va">diagnosis_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_df.html">get_df</a></span><span class="op">(</span><span class="va">diagnosis_list</span>, prefix<span class="op">=</span><span class="st">"ckd"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ~2 seconds</span></span>
<span></span>
<span><span class="co"># how many cases ascertained?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">diagnosis_df</span><span class="op">$</span><span class="va">ckd_bin</span><span class="op">)</span></span>
<span><span class="co">#&gt;      0      1 </span></span>
<span><span class="co">#&gt; 470334  31935 </span></span>
<span></span>
<span><span class="co"># source of earliest diagnosis date</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">diagnosis_df</span><span class="op">$</span><span class="va">ckd_src</span><span class="op">)</span></span>
<span><span class="co">#&gt;    death         gp        hes selfrep_i0 selfrep_i1 selfrep_i2 selfrep_i3 </span></span>
<span><span class="co">#&gt;      224      12394      19310         85         16         63          3</span></span>
<span></span>
<span><span class="co"># date of diagnosis for prevalent cases (i.e., before UK Biobank baseline assessment)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">diagnosis_df</span><span class="op">$</span><span class="va">ckd_df</span><span class="op">[</span> <span class="va">diagnosis_df</span><span class="op">$</span><span class="va">ckd_bin_prev</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;         Min.      1st Qu.       Median         Mean      3rd Qu.         Max. </span></span>
<span><span class="co">#&gt; "1958-01-01" "2006-06-21" "2007-01-12" "2006-06-24" "2007-11-19" "2010-06-16" </span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ascertaining-multiple-conditions-at-once">Ascertaining multiple conditions at once<a class="anchor" aria-label="anchor" href="#ascertaining-multiple-conditions-at-once"></a>
</h3>
<p>The default <code><a href="reference/get_df.html">get_df()</a></code> behaviour is to use all available codes. However the most time-efficient way to get multiple conditions is to run <code><a href="reference/get_diagnoses.html">get_diagnoses()</a></code> once for all codes for the conditions you wish to ascertain, then get the “date first diagnosed” for each condition separately. In the codes data frame you just need a field indicating the condition name, that will become the variable prefixes.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># combine haemochromatosis and CKD codes together</span></span>
<span><span class="co">#   each contain there columns: condition, vocab_id, and code</span></span>
<span><span class="co">#   where `condition` is either "hh" or "ckd" and will become the variable prefix</span></span>
<span><span class="va">codes_df_combined</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu">ukbrapR</span><span class="fu">:::</span><span class="va">codes_df_hh</span>, <span class="fu">ukbrapR</span><span class="fu">:::</span><span class="va">codes_df_ckd</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get diagnosis data - returns list of data frames (one per source)</span></span>
<span><span class="va">diagnosis_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_diagnoses.html">get_diagnoses</a></span><span class="op">(</span><span class="va">codes_df_combined</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># for each participant, get Date First diagnosed with the condition</span></span>
<span><span class="va">diagnosis_df</span> <span class="op">=</span> <span class="fu"><a href="reference/get_df.html">get_df</a></span><span class="op">(</span><span class="va">diagnosis_list</span>, group_by<span class="op">=</span><span class="st">"condition"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># each condition has full set of output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">diagnosis_df</span><span class="op">$</span><span class="va">hh_bin</span><span class="op">)</span></span>
<span><span class="co">#&gt;      0      1 </span></span>
<span><span class="co">#&gt; 500254   2015 </span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">diagnosis_df</span><span class="op">$</span><span class="va">ckd_bin</span><span class="op">)</span></span>
<span><span class="co">#&gt;      0      1 </span></span>
<span><span class="co">#&gt; 470334  31935 </span></span></code></pre></div>
<p>In the above example we also included a UK Biobank self-reported illness code for haemochromatosis, that was also ascertained (the Date First is run on each condition separately, they do not all need to have the same data sources).</p>
</div>
</div>
<div class="section level2">
<h2 id="utility-functions">Utility functions<a class="anchor" aria-label="anchor" href="#utility-functions"></a>
</h2>
<ul>
<li>Check if field IDs are valid and return phenotype names with <code><a href="reference/fields_to_phenos.html">fields_to_phenos()</a></code>
</li>
<li>Label UK Biobank data fields with <code><a href="reference/label_ukb_fields.html">label_ukb_fields()</a></code>
</li>
<li>Upload/download files between worker and RAP with <code><a href="reference/upload_to_rap.html">upload_to_rap()</a></code> and <code><a href="reference/download_from_rap.html">download_from_rap()</a></code>
</li>
<li>Pull phenotypes from Spark instance with <code><a href="reference/get_rap_phenos.html">get_rap_phenos()</a></code>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="questions-and-comments">Questions and comments<a class="anchor" aria-label="anchor" href="#questions-and-comments"></a>
</h2>
<p>Please report any bugs or <a href="https://github.com/lcpilling/ukbrapR/issues" class="external-link">issues</a>, and feel free to suggest changes as <a href="https://github.com/lcpilling/ukbrapR/pulls" class="external-link">pull requests</a>. Alternatively, feel free to contact me via e-mail <a href="mailto:L.Pilling@exeter.ac.uk" class="email">L.Pilling@exeter.ac.uk</a></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcpilling/ukbrapR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/lcpilling/ukbrapR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ukbrapR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://lcpilling.github.io" class="external-link">Luke Pilling</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3332-8454" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcpilling/ukbrapR" class="external-link"><img src="https://img.shields.io/badge/version-0.3.4-informational.svg"></a></li>
<li><a href="https://github.com/lcpilling/ukbrapR/commits/main" class="external-link"><img src="https://img.shields.io/github/last-commit/lcpilling/ukbrapR.svg"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange"></a></li>
<li><a href="https://zenodo.org/doi/10.5281/zenodo.11517716" class="external-link"><img src="https://zenodo.org/badge/709765135.svg" alt="DOI"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://lcpilling.github.io" class="external-link">Luke Pilling</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
