<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>TWIST (Triangulation WIthin A STudy) analysis in R. • twistR</title>
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="TWIST (Triangulation WIthin A STudy) analysis in R.">
<meta name="description" content="An R package to implement the Triangulation WIthin a STudy (TWIST) analysis framework to robusty estimate the Genetically Modified Treatment Effect (GMTE).">
<meta property="og:description" content="An R package to implement the Triangulation WIthin a STudy (TWIST) analysis framework to robusty estimate the Genetically Modified Treatment Effect (GMTE).">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">twistR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.5</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcpilling/twistR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="twistr-">twistR <a href="https://lcpilling.github.io/twistR/"><img src="reference/figures/twistR_hex_v1.png" align="right" width="150"></a>
<a class="anchor" aria-label="anchor" href="#twistr-"></a>
</h1></div>
<p>TWIST (Triangulation WIthin A STudy) analysis in R.</p>
<!-- badges: start -->

<p>The TWIST framework estimates the ‘genetically moderated treatment effect’ (GMTE). This is akin to the population attributable fraction (PAF) in pharmaco-epidemiology: the excess disease or outcomes in individuals prescribed a specific medication that are attributed to genetic variant(s). TWIST extends the PAF by incorporating information on both treated and untreated individuals. With this larger set of information we show that four analysis approaches for estimating the GMTE are possible. Each one relies on a different set of assumptions to work correctly and provides estimates that are largely uncorrelated with one another. A decision framework is used to decide when a particular estimation strategy is most appropriate and how specific estimators can be combined to further improve efficiency. Triangulation of evidence from different data sources, each with their inherent biases and limitations, is becoming a well established principle for strengthening causal analysis. Hence we called our framework ‘Triangulation WIthin a STudy’ (TWIST) in order to emphasise that an analysis in this spirit is also possible within a single data set, using causal estimates that are approximately uncorrelated, but reliant on different sets of assumptions.</p>
<p>TWIST therefore robustly estimates how many events (outcome can be mortality, disease, symptoms, etc) in the genotype group could be avoided if this group could experience the same treatment effect as non-carriers.</p>
<p>Please see our published paper in PLoS Genetics for further information <a href="https://doi.org/10.1371/journal.pgen.1009783" class="external-link uri">https://doi.org/10.1371/journal.pgen.1009783</a></p>
<p>If you use this package please cite</p>
<blockquote>
<p>Bowden J, Pilling L, Türkmen D, Kuo C-L, Melzer D (2021) The Triangulation WIthin a STudy (TWIST) framework for causal inference within pharmacogenetic research. PLoS Genet 17(9):e1009783. <a href="https://doi.org/10.1371/journal.pgen.1009783" class="external-link uri">https://doi.org/10.1371/journal.pgen.1009783</a></p>
</blockquote>
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<p><img align="right" src="reference/figures/TWIST_meme1.png" width="350"></p>
<ul>
<li><a href="#installation">Installation</a></li>
<li><a href="#model-types-performed">Model types performed</a></li>
<li>
<a href="#function-arguments">Function arguments</a>
<ul>
<li><a href="#gmte_binary">gmte_binary()</a></li>
<li><a href="#gmte_continuous">gmte_continuous()</a></li>
<li><a href="#gmte_aalen">gmte_aalen()</a></li>
</ul>
</li>
<li>
<a href="#output-and-examples">Output and examples</a>
<ul>
<li><a href="#example-using-binary-outcome">Example using binary outcome</a></li>
<li><a href="#which-estimate-should-i-use">Which estimate should I use?</a></li>
<li><a href="#what-does-the-estimate-mean">What does the estimate mean?</a></li>
<li><a href="#example-using-aalen-additive-hazards">Example using Aalen additive hazards</a></li>
</ul>
</li>
<li><a href="#forest-plot-of-estimates">Forest plot of estimates</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the development version from GitHub use the <code>remotes</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lcpilling/twistR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To install the latest release, use:</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lcpilling/twistR@*release"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To install a specific version (see tags), use:</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lcpilling/twistR@v0.1.0"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="model-types-performed">Model types performed<a class="anchor" aria-label="anchor" href="#model-types-performed"></a>
</h2>
<p><code>twistR</code> can perform GMTE analysis on three types of outcome: 1. <code><a href="reference/gmte_binary.html">gmte_binary()</a></code> performs logistic regression analysis of a binary outcome (yes/no [1]/[0]). Requires the <a href="https://cran.r-project.org/web/packages/margins/" class="external-link"><code>margins</code></a> package. 2. <code><a href="reference/gmte_continuous.html">gmte_continuous()</a></code> performs linear regression analysis of a continuous outcome (quantitative measure, such as LDL cholesterol). 3. <code><a href="reference/gmte_aalen.html">gmte_aalen()</a></code> performs Aalen additive hazards model i.e. a time-to-event analysis (such as mortality). Requires the <a href="https://cran.r-project.org/web/packages/timereg/" class="external-link"><code>timereg</code></a> package.</p>
</div>
<div class="section level2">
<h2 id="function-arguments">Function arguments<a class="anchor" aria-label="anchor" href="#function-arguments"></a>
</h2>
<div class="section level4">
<h4 id="gmte_binary">gmte_binary()<a class="anchor" aria-label="anchor" href="#gmte_binary"></a>
</h4>
<table class="table">
<colgroup>
<col width="42%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Y</td>
<td>The binary outcome variable name (string) which appears in data.frame <code>D</code>.</td>
</tr>
<tr class="even">
<td>T</td>
<td>The treatment variable name (string) which appears in data.frame <code>D</code>. Assumed to be binary.</td>
</tr>
<tr class="odd">
<td>G</td>
<td>The genotype variable name (string) which appears in data.frame <code>D</code>. Normally binary (e.g. comparing homozygous rare individuals to the rest of the population).</td>
</tr>
<tr class="even">
<td>Z</td>
<td>A string containing the model covariates to appear in the <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> models (for example “age+sex”). All need to be in data.frame <code>D</code>.</td>
</tr>
<tr class="odd">
<td>D</td>
<td>A data.frame containing the above variables.</td>
</tr>
<tr class="even">
<td>Link</td>
<td>Link function for the <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> - needs to be one of “logit”,“probit” or “identity”. If unspecified the default is “logit”.</td>
</tr>
<tr class="odd">
<td>alpha</td>
<td>The p-value threshold for the chi-square test, assessing whether two estimates should be combined. Default is 0.05.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="gmte_continuous">gmte_continuous()<a class="anchor" aria-label="anchor" href="#gmte_continuous"></a>
</h4>
<table class="table">
<colgroup>
<col width="42%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Y</td>
<td>The continuous outcome variable name (string) which appears in data.frame <code>D</code>.</td>
</tr>
<tr class="even">
<td>T</td>
<td>The treatment variable name (string) which appears in data.frame <code>D</code>. Assumed to be binary.</td>
</tr>
<tr class="odd">
<td>G</td>
<td>The genotype variable name (string) which appears in data.frame <code>D</code>. Normally binary (e.g. comparing homozygous rare individuals to the rest of the population).</td>
</tr>
<tr class="even">
<td>Z</td>
<td>A string containing the model covariates to appear in the <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> models (for example “age+sex”). All need to be in data.frame <code>D</code>.</td>
</tr>
<tr class="odd">
<td>D</td>
<td>A data.frame containing the above variables.</td>
</tr>
<tr class="even">
<td>alpha</td>
<td>The p-value threshold for the chi-square test, assessing whether two estimates should be combined. Default is 0.05.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="gmte_aalen">gmte_aalen()<a class="anchor" aria-label="anchor" href="#gmte_aalen"></a>
</h4>
<table class="table">
<colgroup>
<col width="42%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th>Argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Y_t0</td>
<td>Variable name (string) for when participants “enter” the model, which appears in data.frame <code>D</code>. When participants enter the model (can be all 0s if Y_t1 is time since start of exposure). Variable can be in date format from the <code><a href="https://rdrr.io/r/base/as.Date.html" class="external-link">as.Date()</a></code> function, or numeric.</td>
</tr>
<tr class="even">
<td>Y_t1</td>
<td>Variable name (string) for when participants “exit” the model, which appears in data.frame <code>D</code>. Either days since start of exposure period (numeric) or in date format from the <code><a href="https://rdrr.io/r/base/as.Date.html" class="external-link">as.Date()</a></code> function.</td>
</tr>
<tr class="odd">
<td>Y_d</td>
<td>Variable name for the binary “event” variable (string) which appears in data.frame <code>D</code>.</td>
</tr>
<tr class="even">
<td>T</td>
<td>The treatment variable name (string) which appears in data.frame <code>D</code>. Assumed to be binary.</td>
</tr>
<tr class="odd">
<td>G</td>
<td>The genotype variable name (string) which appears in data.frame <code>D</code>. Normally binary (e.g. comparing homozygous rare individuals to the rest of the population).</td>
</tr>
<tr class="even">
<td>Z</td>
<td>A string containing the model covariates to appear in the <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> models (for example “age+sex”). All need to be in data.frame <code>D</code>. Unless otherwise specified covariates will be assumed to be time invarying i.e. the <code>const()</code> wrapper will be added. See <code>aalen()</code> documentation in the <a href="https://cran.r-project.org/web/packages/timereg/" class="external-link"><code>timereg</code></a> package.</td>
</tr>
<tr class="odd">
<td>D</td>
<td>A data.frame containing the above variables.</td>
</tr>
<tr class="even">
<td>Nsim</td>
<td>Number of simulations to perform in the <code>aalen()</code> models. Default is 100.</td>
</tr>
<tr class="odd">
<td>alpha</td>
<td>The p-value threshold for the chi-square test, assessing whether two estimates should be combined. Default is 0.05.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="output-and-examples">Output and examples<a class="anchor" aria-label="anchor" href="#output-and-examples"></a>
</h2>
<p>For each GMTE function a object of class <code>twistR_GMTE</code> is returned. This contains the full model outputs from each individual analysis performed (GMTE1, GMTE0, MR, RGMTE, and CAT – see paper) in addition to a summary of all the models performed and whether the models can be combined to improve estimation in <code>$FullCombined</code>. For example:</p>
<div class="section level4">
<h4 id="example-using-binary-outcome">Example using binary outcome<a class="anchor" aria-label="anchor" href="#example-using-binary-outcome"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example using: </span></span>
<span><span class="co">#   - a binary outcome (high LDL) - in data.frame D</span></span>
<span><span class="co">#   - a binary treatment variable (on statins) - in data.frame D</span></span>
<span><span class="co">#   - a binary genotype (SLCO1B1*5 homozygotes) - in data.frame D</span></span>
<span><span class="co">#   - adjustment for age and genetic principal components of ancestry 1 to 10 - all in data.frame D</span></span>
<span><span class="va">Y</span><span class="op">=</span><span class="st">"ldl_high"</span></span>
<span><span class="cn">T</span><span class="op">=</span><span class="st">"statin"</span></span>
<span><span class="va">G</span><span class="op">=</span><span class="st">"slco1b1_5_hmz"</span></span>
<span><span class="va">Z</span><span class="op">=</span><span class="st">"age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10"</span></span>
<span><span class="va">Link</span><span class="op">=</span><span class="st">"logit"</span></span>
<span><span class="va">results</span><span class="op">=</span><span class="fu"><a href="reference/gmte_binary.html">gmte_binary</a></span><span class="op">(</span><span class="va">Y</span>,<span class="cn">T</span>,<span class="va">G</span>,<span class="va">Z</span>,<span class="va">D</span>,<span class="va">Link</span><span class="op">)</span></span></code></pre></div>
<p>Prints the following results table:</p>
<table class="table">
<colgroup>
<col width="19%">
<col width="9%">
<col width="6%">
<col width="16%">
<col width="16%">
<col width="6%">
<col width="25%">
</colgroup>
<thead><tr class="header">
<th>Model</th>
<th>Est</th>
<th>SE</th>
<th>P.Est</th>
<th>Qstat</th>
<th>Qp</th>
<th>Combine?</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>CAT</td>
<td>-1.315816e+01</td>
<td>0.07672673</td>
<td>0.000000e+00</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>GMTE0</td>
<td>-8.747968e-04</td>
<td>0.00596922</td>
<td>8.834862e-01</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>GMTE1</td>
<td>5.823592e-02</td>
<td>0.01373771</td>
<td>2.243879e-05</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>RGMTE</td>
<td>7.525366e-02</td>
<td>0.01630214</td>
<td>3.908640e-06</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>MR</td>
<td>-8.593626e-04</td>
<td>0.04927101</td>
<td>9.860844e-01</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>RGMTE_MR</td>
<td>6.774350e-02</td>
<td>0.01547698</td>
<td>1.202974e-05</td>
<td>2.150891</td>
<td>0.1424872</td>
<td>1</td>
</tr>
<tr class="odd">
<td>RGMTE_CAT</td>
<td>-4.963459e-01</td>
<td>0.01594618</td>
<td>0.000000e+00</td>
<td>28462.588988</td>
<td>0.0000000</td>
<td>0</td>
</tr>
<tr class="even">
<td>MR_CAT</td>
<td>-3.842414e+00</td>
<td>0.04145881</td>
<td>0.000000e+00</td>
<td>20820.492332</td>
<td>0.0000000</td>
<td>0</td>
</tr>
<tr class="odd">
<td>GMTE1_CAT</td>
<td>-3.522933e-01</td>
<td>0.01352266</td>
<td>0.000000e+00</td>
<td>28749.387485</td>
<td>0.0000000</td>
<td>0</td>
</tr>
<tr class="even">
<td>RGMTE_MR_CAT</td>
<td>-4.493672e-01</td>
<td>0.01517140</td>
<td>0.000000e+00</td>
<td>28554.130703</td>
<td>0.0000000</td>
<td>0</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="which-estimate-should-i-use">Which estimate should I use?<a class="anchor" aria-label="anchor" href="#which-estimate-should-i-use"></a>
</h4>
<p>To understand which estimate is best to use, the user needs to consider the following: * the context of the specific treatment and population, * the assumptions made by each method and whether they are plausible, * whether a pair or triplet of estimates (such as the MR and RGMTE estimates) are sufficiently similar to combine (default alpha 0.05 for Q-statistic (<code>Qp</code>)).</p>
<p>Though the Robust GMTE (RGMTE) estimate is often the most appropriate, in some situations the assumptions may not be satisfied and a combined estimate may be better. For this reason we do not automatically give a recommendation when the analysis is performed. For the combined estimates the <code>Combine?</code> column simply reports whether the p-value for the Q-statistic (<code>Qp</code>) is &gt;0.05 i.e. the estimates do not significantly differ. This does not necesssarily mean it is the best choice. For futher information on the assumptions tested and decision framework please see the published manuscript (open access in PLoS Genetics <a href="https://doi.org/10.1371/journal.pgen.1009783" class="external-link uri">https://doi.org/10.1371/journal.pgen.1009783</a>).</p>
</div>
<div class="section level4">
<h4 id="what-does-the-estimate-mean">What does the estimate mean?<a class="anchor" aria-label="anchor" href="#what-does-the-estimate-mean"></a>
</h4>
<p>We can express the GMTE estimand as the average causal effect if everyone could receive moderated treatment level T* = 1 (i.e. the full or enhanced effect) versus if everyone could receive treatment level T* = 0 (i.e. no enhanced effect).</p>
<ul>
<li>For a binary analysis, the <code>Est</code> is the risk difference if all people could experience the same treatment effect as those with genotype G=1 versus if all people could experience the treatment effect as those with genotype G=0.</li>
<li>For a continuous analysis, the <code>Est</code> is the mean difference in outcome (units) if all people could experience the same treatment effect as those with genotype G=1 versus if all people could experience the treatment effect as those with genotype G=0.</li>
<li>For a time-to-event (Aalen) analysis, the <code>Est</code> is the risk difference per unit time (provided by user in <code>Y_t0</code> and <code>Y_t1</code>, e.g. per year if time coded on the year scale) if all people could experience the same treatment effect as those with genotype G=1 versus if all people could experience the treatment effect as those with genotype G=0.</li>
</ul>
</div>
<div class="section level4">
<h4 id="example-using-aalen-additive-hazards">Example using Aalen additive hazards<a class="anchor" aria-label="anchor" href="#example-using-aalen-additive-hazards"></a>
</h4>
<p>In our paper we include results from our analysis of the anti-platelet drug clopidogrel, where genetic variants in gene <em>CYP2C19</em> impair drug metabolism and reduce function, thereby increasing risk of thrombosis. We used the Aalen additive hazards models to estimate the increased risk of ischemic stroke in UK Biobank participants prescribed clopidogrel, who carried <em>CYP2C19</em> Loss of Function (LoF) variants (<em>2-</em>8).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example using: </span></span>
<span><span class="co">#   - a time-to-event model (time to ischemic stroke) - time to stroke/censoring is in "years" so output should be interpreted this way</span></span>
<span><span class="co">#   - a binary treatment variable (on clopidogrel)</span></span>
<span><span class="co">#   - a binary genotype (CYP2C19 *2-*8 LoF carriers)</span></span>
<span><span class="co">#   - adjustment for age and genetic principal components of ancestry 1 to 10</span></span>
<span><span class="co">#     ^^ all need to be in data.frame D</span></span>
<span><span class="va">Y_t0</span><span class="op">=</span><span class="st">"time_0"</span></span>
<span><span class="va">Y_t1</span><span class="op">=</span><span class="st">"time_to_stroke"</span></span>
<span><span class="va">Y_d</span><span class="op">=</span><span class="st">"stroke_binary"</span></span>
<span><span class="cn">T</span><span class="op">=</span><span class="st">"clopidogrel"</span></span>
<span><span class="va">G</span><span class="op">=</span><span class="st">"cyp2c19_lof"</span></span>
<span><span class="va">Z</span><span class="op">=</span><span class="st">"age+sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10"</span></span>
<span><span class="va">results</span><span class="op">=</span><span class="fu"><a href="reference/gmte_aalen.html">gmte_aalen</a></span><span class="op">(</span><span class="va">Y_t0</span>,<span class="va">Y_t1</span>,<span class="va">Y_d</span>,<span class="cn">T</span>,<span class="va">G</span>,<span class="va">Z</span>,<span class="va">D</span><span class="op">)</span></span></code></pre></div>
<p>Prints the following results table:</p>
<table class="table">
<colgroup>
<col width="20%">
<col width="10%">
<col width="6%">
<col width="13%">
<col width="16%">
<col width="6%">
<col width="26%">
</colgroup>
<thead><tr class="header">
<th>Model</th>
<th>Est</th>
<th>SE</th>
<th>EstP</th>
<th>Qstat</th>
<th>Qp</th>
<th>Combine?</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>CAT</td>
<td>0.021600000</td>
<td>0.0020800000</td>
<td>0.000000e+00</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>GMTE0</td>
<td>-0.000038600</td>
<td>0.0000747000</td>
<td>6.030000e-01</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>GMTE1</td>
<td>0.002770000</td>
<td>0.0014000000</td>
<td>4.550000e-02</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>RGMTE</td>
<td>0.003290000</td>
<td>0.0015800000</td>
<td>3.600000e-02</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="odd">
<td>MR</td>
<td>0.002870000</td>
<td>0.0010800000</td>
<td>7.970000e-03</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
</tr>
<tr class="even">
<td>RGMTE_MR</td>
<td>0.003003747</td>
<td>0.0008916086</td>
<td>7.546674e-04</td>
<td>0.04815988</td>
<td>8.262967e-01</td>
<td>1</td>
</tr>
<tr class="odd">
<td>RGMTE_CAT</td>
<td>0.009989461</td>
<td>0.0012581694</td>
<td>1.998401e-15</td>
<td>49.13761212</td>
<td>2.386202e-12</td>
<td>0</td>
</tr>
<tr class="even">
<td>MR_CAT</td>
<td>0.006847329</td>
<td>0.0009584958</td>
<td>9.077183e-13</td>
<td>63.86777236</td>
<td>1.332268e-15</td>
<td>0</td>
</tr>
<tr class="odd">
<td>GMTE1_CAT</td>
<td>0.008640896</td>
<td>0.0011614228</td>
<td>1.008083e-13</td>
<td>56.40253563</td>
<td>5.906386e-14</td>
<td>0</td>
</tr>
<tr class="even">
<td>RGMTE_MR_CAT</td>
<td>0.005890357</td>
<td>0.0008194919</td>
<td>6.583623e-13</td>
<td>67.57323818</td>
<td>2.109424e-15</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>(This table appears as Table 5 in the paper in PLoS Genetics <a href="https://doi.org/10.1371/journal.pgen.1009783" class="external-link uri">https://doi.org/10.1371/journal.pgen.1009783</a>)</p>
<p><strong>Estimating the GMTE / interpreting the output</strong></p>
<p>To estimate the GMTE in this case we modelled the time to stroke using an Aalen additive hazards model, as described in the paper. All models were adjusted for age at recruitment or first Clopidogrel prescription, sex, and the first 10 genetic principal components of ancestry. The above Table show the results for this analysis, which reflect the genetically moderated effect of Clopidogrel treatment on the hazard of stroke <em>per year</em>, expressed as a proportion (i.e. multiply by 100 to get percentage, as shown in Table 5 of paper). The GMTE(1) estimate suggests that being a <em>CYP2C19</em> LoF carrier (G = 1) increases the risk of stroke by 0.28% (p = 0.048) compared to those without the LoF variant (G = 0). To put this figure in context, if we could reduce the LoF carrier’s risk by this amount then, when multiplied by the 5264 LoF carrier patient years in the data, it would lead to an expected 13.2% reduction in the total number of strokes. There were 110 strokes in the LoF group, so we would expect this to reduce by 15 (110 to 95) if carriers of a <em>CYP2C19</em> LoF variant could experience the same treatment effect as non-carriers.</p>
<p>To test for potential bias in the GMTE(1) estimate, we calculate the GMTE(0) estimate in the untreated population. Thankfully, it is close to zero (Hazard diff = -0.0039%, p = 0.61), although slightly negative. Taken at face value, this suggest LoF carriers have a slightly reduced risk of stroke through pathways other than Clopidogrel use. Next we calculate the Corrected As Treated (CAT) estimate. As discussed in the paper, the validity of this method rests strongly on being able to identify all confounders of Clopidogrel use and stroke. With the data available, it was only possible to adjust for age, sex and genetic principal components and perhaps unsurprisingly, the CAT estimate is an order of magnitude larger (Hazard diff = 2.2%, p=2*10−16). Consequently, the Q[CAT,GMTE(1)] statistic detects large heterogeneity and suggests that the CAT and GMTE(1) estimates should not be combined.</p>
<p>For completeness, we next calculate the RGMTE estimate for the GMTE hazard difference. Since this is itself the difference between the GMTE(1) and GMTE(0) estimates, and given they are of opposite sign, the RMGTE estimate is slightly larger at 0.33% (p = 0.037), suggesting 17 strokes could have been avoided. The MR estimate for the GMTE hazard difference is similar at 0.29% (p = 0.008). Heterogeneity analysis reveals that the MR and RGMTE estimates are sufficiently similar to combine into a more precise single estimate of the GMTE (Q[MR,RGMTE] = 0.8). The combined estimate is 0.3 (p = 7.5*10-4), or that 16 strokes could have been avoided. No other combination of estimates are sufficiently similar to combine.</p>
</div>
</div>
<div class="section level2">
<h2 id="forest-plot-of-estimates">Forest plot of estimates<a class="anchor" aria-label="anchor" href="#forest-plot-of-estimates"></a>
</h2>
<p>To create a simple Forest Plot the estimates simply provide the output of a GMTE analysis:</p>
<pre class="rgmte"><code><span><span class="co"># Example using a continuous outcome (LDL), binary treatment (statins), and binary genotype (SLCO1B1*5 homozygotes) variables</span></span>
<span><span class="va">Y</span><span class="op">=</span><span class="st">"ldl"</span></span>
<span><span class="cn">T</span><span class="op">=</span><span class="st">"statin"</span></span>
<span><span class="va">G</span><span class="op">=</span><span class="st">"slco1b1_5_hmz"</span></span>
<span><span class="va">Z</span><span class="op">=</span><span class="st">"age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10"</span></span>
<span><span class="va">results</span><span class="op">=</span><span class="fu"><a href="reference/gmte_continuous.html">gmte_continuous</a></span><span class="op">(</span><span class="va">Y</span>,<span class="cn">T</span>,<span class="va">G</span>,<span class="va">Z</span>,<span class="va">D</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Model                 Est          SE         EstP        Qstat        Qp Combine?</span></span>
<span><span class="co">#CAT          -47.79761593 0.156521799 0.000000e+00           NA        NA       NA</span></span>
<span><span class="co">#GMTE0         -0.02057607 0.008486743 1.533007e-02           NA        NA       NA</span></span>
<span><span class="co">#GMTE1          0.06984053 0.020476099 6.477051e-04           NA        NA       NA</span></span>
<span><span class="co">#RGMTE          0.09039080 0.022162475 4.532468e-05           NA        NA       NA</span></span>
<span><span class="co">#MR            -0.01003960 0.058701522 8.642019e-01           NA        NA       NA</span></span>
<span><span class="co">#RGMTE_MR       0.07786137 0.020733971 1.731645e-04     2.561888 0.1094678        1</span></span>
<span><span class="co">#RGMTE_CAT     -0.85083255 0.021943596 0.000000e+00 91766.240141 0.0000000        0</span></span>
<span><span class="co">#MR_CAT        -5.90268766 0.054963272 0.000000e+00 81719.705349 0.0000000        0</span></span>
<span><span class="co">#GMTE1_CAT     -0.73556685 0.020303105 0.000000e+00 91952.071936 0.0000000        0</span></span>
<span><span class="co">#RGMTE_MR_CAT  -0.74774632 0.020554416 0.000000e+00 91946.240796 0.0000000        0</span></span>
<span></span>
<span><span class="co"># plot these estimates, highlighting the individual and combined estimates</span></span>
<span><span class="fu"><a href="reference/gmte_plot.html">gmte_plot</a></span><span class="op">(</span><span class="va">results</span>, plot_title <span class="op">=</span> <span class="st">"SLCO1B1*5 effect on LDL during statin treatment"</span><span class="op">)</span></span></code></pre>
<p><img src="reference/figures/gmte_plot.1.png" width="650"></p>
<div class="section level3">
<h3 id="remove-cat-estimates-for-cleaner-plot">Remove CAT estimates for “cleaner” plot<a class="anchor" aria-label="anchor" href="#remove-cat-estimates-for-cleaner-plot"></a>
</h3>
<p>As previously mentioned, the validity of the CAT method rests strongly on being able to identify all confounders of the treatment and the outcome. In this analysis it was only possible to adjust for age, sex and genetic principal components. Therefore the CAT estimate is disproportionately larger and consequently, the Q[CAT,GMTE(1)] statistic detects large heterogeneity, suggesting that the CAT and GMTE(1) estimates should not be combined.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/gmte_plot.html">gmte_plot</a></span><span class="op">(</span><span class="va">results</span>, plot_title <span class="op">=</span> <span class="st">"SLCO1B1*5 effect on LDL during statin treatment"</span>, plot_cat<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/gmte_plot.2.png" width="650"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcpilling/twistR/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing twistR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://lcpilling.github.io" class="external-link">Luke Pilling</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3332-8454" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jack Bowden <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2628-3304" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcpilling/twistR" class="external-link"><img src="https://img.shields.io/badge/version-0.1.5-informational.svg"></a></li>
<li><a href="https://github.com/lcpilling/twistR/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/lcpilling/twistR.svg"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange"></a></li>
<li><a href="https://zenodo.org/badge/latestdoi/402818137" class="external-link"><img src="https://zenodo.org/badge/402818137.svg" alt="DOI"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://lcpilling.github.io" class="external-link">Luke Pilling</a>, Jack Bowden.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
