<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Some useful R functions for processing GWAS output. Some shamelessly borrowed from other packages and the internet generally.">
<title>Some useful R functions for processing GWAS output • gwasRtools</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Some useful R functions for processing GWAS output">
<meta property="og:description" content="Some useful R functions for processing GWAS output. Some shamelessly borrowed from other packages and the internet generally.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">gwasRtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lukepilling/gwasRtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gwasrtools">gwasRtools<a class="anchor" aria-label="anchor" href="#gwasrtools"></a>
</h1></div>
<p>Some useful R functions for processing GWAS output</p>
<p><a href="https://github.com/lukepilling/gwasRtools" class="external-link"><img src="https://img.shields.io/badge/version-0.1.2-informational.svg"></a> <a href="https://github.com/lukepilling/gwasRtools/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/lukepilling/gwasRtools.svg"></a> <a href="https://www.tidyverse.org/lifecycle/#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange"></a> <a href="https://zenodo.org/badge/latestdoi/655790727" class="external-link"><img src="https://zenodo.org/badge/655790727.svg" alt="DOI"></a></p>
<div class="section level2">
<h2 id="list-of-functions">List of functions<a class="anchor" aria-label="anchor" href="#list-of-functions"></a>
</h2>
<ul>
<li><a href="#lambda_gc">lambda_gc()</a></li>
<li><a href="#get_loci">get_loci()</a></li>
<li><a href="#get_nearest_gene">get_nearest_gene()</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install <code>gwasRtools</code> from GitHub use the <code>remotes</code> package:</p>
<p><code>remotes::install_github("lukepilling/gwasRtools")</code></p>
<p>To update the package just run the above command again.</p>
</div>
<div class="section level2">
<h2 id="lambda_gc">lambda_gc()<a class="anchor" aria-label="anchor" href="#lambda_gc"></a>
</h2>
<p>Estimate inflation of test statistics. Lambda GC compares the median test statistic against the expected median test statistic under the null hypothesis of no association. For well-powered quantitative traits with a known polygenic inheritance, we expect inflation of lambda GC, but for traits with no expected association, we expect lambda GC to be around 1.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/lambda_gc.html">lambda_gc</a></span><span class="op">(</span><span class="va">p_values</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get_loci">get_loci()<a class="anchor" aria-label="anchor" href="#get_loci"></a>
</h2>
<p>Determine loci from a GWAS summary statistics file. Use distance from lead significant SNP to estimate independet loci in GWAS summary stats. Uses -log10(p) derived from BETA/SE so does not need P as input. Example below with default input (output truncated):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_loci.html">get_loci</a></span><span class="op">(</span></span>
<span>  <span class="va">gwas</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  maf_col <span class="op">=</span> <span class="st">"MAF"</span>,</span>
<span>  beta_col <span class="op">=</span> <span class="st">"BETA"</span>,</span>
<span>  se_col <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>  n_bases <span class="op">=</span> <span class="fl">5e5</span>,</span>
<span>  p_threshold <span class="op">=</span> <span class="fl">5e-8</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># example using BOLT-LMM output:</span></span>
<span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu"><a href="reference/get_loci.html">get_loci</a></span><span class="op">(</span><span class="va">gwas</span>, maf_col<span class="op">=</span><span class="st">"A1FREQ"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span>, <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 5 × 11</span></span>
<span><span class="co">#&gt;    SNP               CHR        BP ALLELE1 ALLELE0 A1FREQ    BETA      SE  P_BOLT_LMM locus lead </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;           &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;lgl&gt;</span></span>
<span><span class="co">#&gt;  1 rs3041466           2 142820625 C       CAA      0.492 -0.0177 0.00317 0.000000021     1 TRUE </span></span>
<span><span class="co">#&gt;  2 rs5858140           4  49039586 C       CA       0.429 -0.0182 0.00332 0.000000036     2 FALSE</span></span>
<span><span class="co">#&gt;  3 rs2605231           4  49057424 A       T        0.293 -0.0179 0.00329 0.000000044     2 FALSE</span></span>
<span><span class="co">#&gt;  4 4:49057608_AT_A     4  49057608 AT      A        0.301 -0.0188 0.00338 0.000000024     2 FALSE</span></span>
<span><span class="co">#&gt;  5 4:49057930_CT_C     4  49057930 CT      C        0.286 -0.0186 0.00336 0.000000029     2 FALSE</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">lead</span><span class="op">==</span><span class="cn">TRUE</span><span class="op">)</span>, <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 5 × 11</span></span>
<span><span class="co">#&gt;    SNP          CHR        BP ALLELE1 ALLELE0 A1FREQ    BETA      SE P_BOLT_LMM locus lead </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;lgl&gt;</span></span>
<span><span class="co">#&gt;  1 rs3041466      2 142820625 C       CAA      0.492 -0.0177 0.00317   2.10e- 8     1 TRUE </span></span>
<span><span class="co">#&gt;  2 rs11722559     4  49227587 T       C        0.225 -0.0240 0.00391   7.70e-10     2 TRUE </span></span>
<span><span class="co">#&gt;  3 rs4865414      4  52758294 C       T        0.684  0.0200 0.00322   5.10e-10     3 TRUE </span></span>
<span><span class="co">#&gt;  4 rs74405522     6 157205286 T       C        0.962  0.0432 0.00788   4.20e- 8     4 TRUE </span></span>
<span><span class="co">#&gt;  5 rs55730499     6 161005610 C       T        0.918  0.0378 0.00548   5.80e-12     5 TRUE</span></span></code></pre></div>
<ul>
<li>Loci are numbered. Variants within a locus (i.e., significant below the <code>p_threshold</code> and less than <code>n_bases</code> from last significant variant).</li>
<li>Lead variant for each locus is highlighted where <code>lead==TRUE</code> (i.e., smallest p-value for any variant within a locus)</li>
</ul>
<div class="section level3">
<h3 id="use-ld-clumping-to-identify-independent-snps-at-the-same-locus">Use LD clumping to identify independent SNPs at the same locus<a class="anchor" aria-label="anchor" href="#use-ld-clumping-to-identify-independent-snps-at-the-same-locus"></a>
</h3>
<p>Setting option <code>get_ld_indep=TRUE</code> will use {<a href="https://github.com/MRCIEU/ieugwasr" class="external-link">ieugwasr</a>} package <code>ld_clump()</code> function to run Plink LD clumping.</p>
<p>Default is to use a local Plink installation (this is faster) with EUR reference panel. But setting option <code>ld_clump_local</code> to FALSE will use the online IEU API. See the {ieugwasr} docs for details. Default R2 threshold for LD pruning is 0.001 (modify with <code>ld_pruning_r2</code> option).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_loci.html">get_loci</a></span><span class="op">(</span></span>
<span>  <span class="va">gwas</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  maf_col <span class="op">=</span> <span class="st">"MAF"</span>,</span>
<span>  beta_col <span class="op">=</span> <span class="st">"BETA"</span>,</span>
<span>  se_col <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>  n_bases <span class="op">=</span> <span class="fl">5e5</span>,</span>
<span>  p_threshold <span class="op">=</span> <span class="fl">5e-8</span>,</span>
<span>  get_ld_indep<span class="op">=</span><span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># example using BOLT-LMM output:</span></span>
<span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu"><a href="reference/get_loci.html">get_loci</a></span><span class="op">(</span><span class="va">gwas</span>, maf_col<span class="op">=</span><span class="st">"A1FREQ"</span>, get_ld_indep<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ld_clump</span><span class="op">==</span><span class="cn">TRUE</span><span class="op">)</span>, <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 5 × 11</span></span>
<span><span class="co">#&gt;   SNP           CHR        BP ALLELE1 ALLELE0 A1FREQ    BETA      SE P_BOLT_LMM locus lead  ld_indep</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;lgl&gt; &lt;lgl&gt;</span></span>
<span><span class="co">#&gt; 1 rs333957        1 110434791 C       G       0.579   0.0432 0.00708  9.2 e- 10     1 TRUE  TRUE</span></span>
<span><span class="co">#&gt; 2 rs182541539     2 189399724 C       T       0.993  -0.270  0.0459   1   e-  8     2 FALSE TRUE</span></span>
<span><span class="co">#&gt; 3 rs10207004      2 190246799 C       T       0.964  -0.356  0.0182   1.20e- 88     2 FALSE TRUE</span></span>
<span><span class="co">#&gt; 4 rs78842559      2 190396563 G       T       0.977  -0.221  0.0228   3.20e- 23     2 FALSE TRUE</span></span>
<span><span class="co">#&gt; 5 rs4428180       3 133466374 A       G       0.851  -0.120  0.00965  7.5 e- 38     3 TRUE  TRUE</span></span></code></pre></div>
<p>Where before, locus 2 would only have had one “lead” SNP (the lowest p-value) <code>ld_clump()</code> has identified multiple independent variants in the region.</p>
<p>Note that the original <code>locus</code> and <code>lead</code> columns remain, but a new column <code>ld_indep</code> is added indicating the <code>ld_clump()</code> identified lead SNPs. Not necessarily including the original <code>lead</code> especially if not all GWAS SNPs are in the reference panel.</p>
</div>
</div>
<div class="section level2">
<h2 id="get_nearest_gene">get_nearest_gene()<a class="anchor" aria-label="anchor" href="#get_nearest_gene"></a>
</h2>
<p>Get nearest gene from a set of variants using GENCODE data. Need to provide a data.frame of variant IDs (e.g., rsids), CHR and POS. Defaults below, with example output:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_nearest_gene.html">get_nearest_gene</a></span><span class="op">(</span></span>
<span>  <span class="va">snps</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  build   <span class="op">=</span> <span class="fl">37</span>,</span>
<span>  n_bases <span class="op">=</span> <span class="fl">1e5</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 3 × 5</span></span>
<span><span class="co">#&gt;   SNP          CHR        BP gene        dist</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 rs55730499     6 161005610 LPA       -53095</span></span>
<span><span class="co">#&gt; 3 rs814573      19  45424351 APOC1       1745</span></span>
<span><span class="co">#&gt; 3 rs123456      20  98765432 NA            NA</span></span></code></pre></div>
<ul>
<li>If <code>dist</code> is positive, the variant is intergenic, and this is the distance to the closest gene.</li>
<li>If <code>dist</code> is negative, the variant is within a gene, and this is the distance to the start of the gene.</li>
<li>If <code>dist</code> is NA, the variant is not within <code>n_bases</code> of a gene in GENCODE.</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lukepilling/gwasRtools/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/lukepilling/gwasRtools/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gwasRtools</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Luke Pilling <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3332-8454" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Luke Pilling.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
