<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Some useful R functions for processing GWAS output. Some shamelessly borrowed from other packages and the internet generally.">
<title>Some useful R functions for processing GWAS output • gwasRtools</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Some useful R functions for processing GWAS output">
<meta property="og:description" content="Some useful R functions for processing GWAS output. Some shamelessly borrowed from other packages and the internet generally.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">gwasRtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lukepilling/gwasRtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gwasrtools-">gwasRtools <a href="https://lukepilling.github.io/gwasRtools/" class="external-link"><img src="https://raw.githubusercontent.com/lukepilling/gwasRtools/master/images/gwasRtools.png" align="right" width="150"></a>
<a class="anchor" aria-label="anchor" href="#gwasrtools-"></a>
</h1></div>
<p>Some useful R functions for processing GWAS output</p>
<!-- badges: start -->

<div class="section level2">
<h2 id="list-of-functions">List of functions<a class="anchor" aria-label="anchor" href="#list-of-functions"></a>
</h2>
<ul>
<li><a href="#get_loci">get_loci()</a></li>
<li><a href="#get_nearest_gene">get_nearest_gene()</a></li>
<li><a href="#lambda_gc">lambda_gc()</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the development version from GitHub use the <code>remotes</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lukepilling/gwasRtools"</span><span class="op">)</span></span></code></pre></div>
<p>The development version may not “work.” Install a “release” version if needed.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To install the latest release, use:</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lukepilling/gwasRtools@*release"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To install a specific version (see tags), use:</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lukepilling/gwasRtools@v0.1.3"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-dataset">Example dataset<a class="anchor" aria-label="anchor" href="#example-dataset"></a>
</h2>
<p>The package includes a subset of variants from the Graham et al. 2021 GWAS of LDL in 1,320,016 Europeans (GWAS catalog GCST90239658). I will use this throughout.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lukepilling.github.io/gwasRtools" class="external-link">gwasRtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_example</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SNP CHR        BP    A1 A2      MAF        BETA        SE     P</span></span>
<span><span class="co">#&gt; 1 rs370704455   1 104275751 CATCT  C 7.21e-02 -0.00350057 0.0040782 0.391</span></span>
<span><span class="co">#&gt; 2 rs545633289   1 104275758     C  G 5.25e-04 -0.36329200 0.2903640 0.211</span></span>
<span><span class="co">#&gt; 3 rs367651119   1 104275797     G  A 5.99e-03 -0.01142350 0.0164861 0.488</span></span>
<span><span class="co">#&gt; 4 rs560138913   1 104296901     T  G 3.67e-03 -0.05503450 0.0318479 0.084</span></span>
<span><span class="co">#&gt; 5 rs533558585   1 104296979     T  C 7.57e-05  0.00173860 0.1826840 0.992</span></span>
<span><span class="co">#&gt; 6 rs551694811   1 104297001     T  A 1.97e-03 -0.01075230 0.0316984 0.734</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get_loci">get_loci()<a class="anchor" aria-label="anchor" href="#get_loci"></a>
</h2>
<p>Determine loci from GWAS summary statistics. Use distance from lead significant SNP to estimate independent loci [default distance = 500kb]. Uses -log10(p) derived from BETA/SE so does not need P as input. Example below with default input.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu"><a href="reference/get_loci.html">get_loci</a></span><span class="op">(</span><span class="va">gwas_example</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Locus size (bases) = 5e+05</span></span>
<span><span class="co">#&gt; P-value threshold = 5e-08</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; N variants = 319732</span></span>
<span><span class="co">#&gt; N variants p&lt;threshold = 4132</span></span>
<span><span class="co">#&gt; N loci = 15</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span><span class="op">)</span></span>
<span><span class="co">#&gt;               SNP CHR        BP   A1 A2   MAF   BETA     SE        P locus  lead</span></span>
<span><span class="co">#&gt; 57882  rs12046439   1 107536799    T  C 0.248 0.0099 0.0017 5.01e-09     1 FALSE</span></span>
<span><span class="co">#&gt; 57900 rs143849791   1 107537916 CATG  C 0.325 0.0128 0.0016 5.85e-15     1 FALSE</span></span>
<span><span class="co">#&gt; 57922 rs113329442   1 107539252    A  G 0.330 0.0110 0.0014 1.27e-13     1 FALSE</span></span>
<span><span class="co">#&gt; 57987   rs3861909   1 107544176    G  A 0.327 0.0118 0.0015 3.52e-15     1 FALSE</span></span>
<span><span class="co">#&gt; 58025  rs17496332   1 107546375    A  G 0.331 0.0111 0.0014 8.70e-14     1 FALSE</span></span>
<span><span class="co">#&gt; 58091   rs2878349   1 107549245    G  A 0.327 0.0118 0.0014 2.33e-15     1 FALSE</span></span>
<span></span>
<span><span class="va">gwas_loci</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">lead</span><span class="op">==</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SNP CHR        BP A1 A2   MAF    BETA     SE         P locus lead</span></span>
<span><span class="co">#&gt; 1 rs111232683   1 107566149  G  C 0.343  0.0135 0.0016  5.43e-17     1 TRUE</span></span>
<span><span class="co">#&gt; 2 rs114254196   1 108635400  C  T 0.008 -0.0448 0.0081  4.38e-08     2 TRUE</span></span>
<span><span class="co">#&gt; 3 rs115292790   1 109310728  G  A 0.013 -0.0563 0.0060  2.01e-20     3 TRUE</span></span>
<span><span class="co">#&gt; 4  rs12740374   1 109817590  G  T 0.219 -0.1482 0.0016 4.73e-305     4 TRUE</span></span>
<span><span class="co">#&gt; 5 rs140266316   1 110326545  G  A 0.016 -0.0577 0.0059  4.73e-22     5 TRUE</span></span>
<span><span class="co">#&gt; 6    rs657801   1 111736389  T  C 0.315  0.0090 0.0015  1.88e-09     6 TRUE</span></span></code></pre></div>
<ul>
<li>Loci are numbered. Variants within a locus (i.e., significant below the <code>p_threshold</code> and less than <code>n_bases</code> from last significant variant).</li>
<li>Lead variant for each locus is highlighted where <code>lead==TRUE</code> (i.e., smallest p-value for any variant within a locus)</li>
</ul>
<p>The HLA region can be treated as one continuous locus by setting <code>single_hla_locus</code> to TRUE.</p>
<div class="section level3">
<h3 id="use-ld-clumping-to-identify-independent-snps-at-the-same-locus">Use LD clumping to identify independent SNPs at the same locus<a class="anchor" aria-label="anchor" href="#use-ld-clumping-to-identify-independent-snps-at-the-same-locus"></a>
</h3>
<p>Setting option <code>ld_clump=TRUE</code> will use {<a href="https://github.com/MRCIEU/ieugwasr" class="external-link">ieugwasr</a>} package <code>ld_clump()</code> function to run Plink LD clumping.</p>
<p>Default is to use a local Plink installation (this is faster) with EUR reference panel. But setting option <code>ld_clump_local</code> to FALSE will use the online IEU API. See the {ieugwasr} docs for details. Default R2 threshold for LD pruning is 0.01 (modify with <code>ld_pruning_r2</code> option).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu"><a href="reference/get_loci.html">get_loci</a></span><span class="op">(</span><span class="va">gwas_example</span>, ld_clump<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ** Performing LD clumping. Can take a few minutes</span></span>
<span><span class="co">#&gt; ** Local Plink installation will be called -- output appears in your R terminal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Locus size (bases) = 5e+05</span></span>
<span><span class="co">#&gt; P-value threshold = 5e-08</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; N variants = 319732</span></span>
<span><span class="co">#&gt; N variants p&lt;threshold = 4132</span></span>
<span><span class="co">#&gt; N loci = 15</span></span>
<span><span class="co">#&gt; N independent variants (LD R2 threshold 0.01) = 153</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span><span class="op">)</span></span>
<span><span class="co">#&gt;               SNP CHR        BP   A1 A2   MAF   BETA     SE        P locus  lead lead_dist lead_ld</span></span>
<span><span class="co">#&gt; 57882  rs12046439   1 107536799    T  C 0.248 0.0099 0.0017 5.01e-09     1 FALSE     FALSE   FALSE</span></span>
<span><span class="co">#&gt; 57900 rs143849791   1 107537916 CATG  C 0.325 0.0128 0.0016 5.85e-15     1 FALSE     FALSE   FALSE</span></span>
<span><span class="co">#&gt; 57922 rs113329442   1 107539252    A  G 0.330 0.0110 0.0014 1.27e-13     1 FALSE     FALSE   FALSE</span></span>
<span><span class="co">#&gt; 57987   rs3861909   1 107544176    G  A 0.327 0.0118 0.0015 3.52e-15     1 FALSE     FALSE   FALSE</span></span>
<span><span class="co">#&gt; 58025  rs17496332   1 107546375    A  G 0.331 0.0111 0.0014 8.70e-14     1 FALSE     FALSE   FALSE</span></span>
<span><span class="co">#&gt; 58091   rs2878349   1 107549245    G  A 0.327 0.0118 0.0014 2.33e-15     1 FALSE     FALSE   FALSE</span></span>
<span></span>
<span><span class="va">gwas_loci</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">lead</span><span class="op">==</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SNP CHR        BP A1 A2   MAF    BETA     SE        P locus lead lead_dist lead_ld</span></span>
<span><span class="co">#&gt; 1 rs111232683   1 107566149  G  C 0.343  0.0135 0.0016 5.43e-17     1 TRUE      TRUE    TRUE</span></span>
<span><span class="co">#&gt; 2 rs114254196   1 108635400  C  T 0.008 -0.0448 0.0081 4.38e-08     2 TRUE      TRUE    TRUE</span></span>
<span><span class="co">#&gt; 3 rs140300970   1 109020060  A  T 0.022 -0.0278 0.0049 2.13e-08     3 TRUE     FALSE    TRUE</span></span>
<span><span class="co">#&gt; 4 rs148503795   1 109166178  C  G 0.010 -0.0423 0.0070 2.47e-09     3 TRUE     FALSE    TRUE</span></span>
<span><span class="co">#&gt; 5  rs74896173   1 109167705  T  C 0.009 -0.0429 0.0076 2.08e-08     3 TRUE     FALSE    TRUE</span></span>
<span><span class="co">#&gt; 6 rs111751551   1 109242056  G  A 0.010 -0.0505 0.0069 4.32e-13     3 TRUE     FALSE    TRUE</span></span></code></pre></div>
<p>Where before, locus 3 would only have had one lead SNP (based on distance/lowest p-value) <code>ld_clump()</code> has identified multiple independent variants in the region.</p>
<p>Note that there are now three <code>lead</code> columns: - <code>lead_dist</code> is the original <code>lead</code> column, simply based on distance and p-values - <code>lead_ld</code> is the direct results from <code>ld_clump()</code> - The <code>lead</code> column combines the two (some SNPs are missing from LD panel so it does not always choose the lowest p-value if only 1 variant identified at a locus).</p>
<p>** Note that <code>ld_clump()</code> only considers R^2 when defining independent variants, not D’ – you should perform additional checking/conditional analysis where relevant for <code>lead</code> variants in close proximity.</p>
</div>
</div>
<div class="section level2">
<h2 id="get_nearest_gene">get_nearest_gene()<a class="anchor" aria-label="anchor" href="#get_nearest_gene"></a>
</h2>
<p>Get nearest gene from a set of variants using GENCODE data. Need to provide a data.frame of variant IDs (e.g., rsids), CHR and POS. Default column names are the same as for <code><a href="reference/get_loci.html">get_loci()</a></code>. Default max distance from variant to gene is 100kb.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu"><a href="reference/get_nearest_gene.html">get_nearest_gene</a></span><span class="op">(</span><span class="va">gwas_loci</span>, build<span class="op">=</span><span class="fl">37</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using human genome build 37</span></span>
<span><span class="co">#&gt; Getting nearest gene for 4131 unique variants</span></span>
<span><span class="co">#&gt; (Removed 1 duplicated or missing variant IDs/positions)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SNP CHR        BP   A1 A2   MAF   BETA     SE        P locus  lead  gene  dist</span></span>
<span><span class="co">#&gt; 1  rs12046439   1 107536799    T  C 0.248 0.0099 0.0017 5.01e-09     1 FALSE PRMT6 62468</span></span>
<span><span class="co">#&gt; 2 rs143849791   1 107537916 CATG  C 0.325 0.0128 0.0016 5.85e-15     1 FALSE PRMT6 61351</span></span>
<span><span class="co">#&gt; 3 rs113329442   1 107539252    A  G 0.330 0.0110 0.0014 1.27e-13     1 FALSE PRMT6 60015</span></span>
<span><span class="co">#&gt; 4   rs3861909   1 107544176    G  A 0.327 0.0118 0.0015 3.52e-15     1 FALSE PRMT6 55091</span></span>
<span><span class="co">#&gt; 5  rs17496332   1 107546375    A  G 0.331 0.0111 0.0014 8.70e-14     1 FALSE PRMT6 52892</span></span>
<span><span class="co">#&gt; 6   rs2878349   1 107549245    G  A 0.327 0.0118 0.0014 2.33e-15     1 FALSE PRMT6 50022</span></span>
<span></span>
<span><span class="va">gwas_loci</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">lead</span><span class="op">==</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SNP CHR        BP A1 A2   MAF    BETA     SE        P locus lead     gene   dist</span></span>
<span><span class="co">#&gt; 1 rs111232683   1 107566149  G  C 0.343  0.0135 0.0016 5.43e-17     1 TRUE    PRMT6  33118</span></span>
<span><span class="co">#&gt; 2 rs114254196   1 108635400  C  T 0.008 -0.0448 0.0081 4.38e-08     2 TRUE SLC25A24  41258</span></span>
<span><span class="co">#&gt; 3 rs140300970   1 109020060  A  T 0.022 -0.0278 0.0049 2.13e-08     3 TRUE    NBPF6   6436</span></span>
<span><span class="co">#&gt; 4 rs148503795   1 109166178  C  G 0.010 -0.0423 0.0070 2.47e-09     3 TRUE  FAM102B -63467</span></span>
<span><span class="co">#&gt; 5  rs74896173   1 109167705  T  C 0.009 -0.0429 0.0076 2.08e-08     3 TRUE  FAM102B -64994</span></span>
<span><span class="co">#&gt; 6 rs111751551   1 109242056  G  A 0.010 -0.0505 0.0069 4.32e-13     3 TRUE  PRPF38B  -7111</span></span></code></pre></div>
<ul>
<li>If <code>dist</code> is positive, the variant is intergenic, and this is the distance to the closest gene.</li>
<li>If <code>dist</code> is negative, the variant is within a gene, and this is the distance to the start of the gene.</li>
<li>If <code>dist</code> is NA, the variant is not within <code>n_bases</code> of a gene in GENCODE.</li>
</ul>
</div>
<div class="section level2">
<h2 id="lambda_gc">lambda_gc()<a class="anchor" aria-label="anchor" href="#lambda_gc"></a>
</h2>
<p>Estimate inflation of test statistics. Lambda GC compares the median test statistic against the expected median test statistic under the null hypothesis of no association. For well-powered GWAS of traits with a known polygenic inheritance, we expect inflation of lambda GC. For traits with no expected association, we expect lambda GC to be around 1.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/lambda_gc.html">lambda_gc</a></span><span class="op">(</span><span class="va">gwas_example</span><span class="op">$</span><span class="va">P</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1.41112</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lukepilling/gwasRtools/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/lukepilling/gwasRtools/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gwasRtools</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://lukepilling.github.io" class="external-link">Luke Pilling</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3332-8454" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lukepilling/gwasRtools" class="external-link"><img src="https://img.shields.io/badge/version-0.1.6-informational.svg"></a></li>
<li><a href="https://github.com/lukepilling/gwasRtools/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/lukepilling/gwasRtools.svg"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange"></a></li>
<li><a href="https://zenodo.org/badge/latestdoi/655790727" class="external-link"><img src="https://zenodo.org/badge/655790727.svg" alt="DOI"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://lukepilling.github.io" class="external-link">Luke Pilling</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
