<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Use distance variant position to get the nearest gene. Uses snpsettest function `map_snp_to_gene` to identify genes from GENCODE databases (https://github.com/HimesGroup/snpsettest)."><title>Get nearest gene from a set of variants — get_nearest_gene • gwasRtools</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get nearest gene from a set of variants — get_nearest_gene"><meta property="og:description" content="Use distance variant position to get the nearest gene. Uses snpsettest function `map_snp_to_gene` to identify genes from GENCODE databases (https://github.com/HimesGroup/snpsettest)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gwasRtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lukepilling/gwasRtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Get nearest gene from a set of variants</h1>
      <small class="dont-index">Source: <a href="https://github.com/lukepilling/gwasRtools/blob/HEAD/R/get_nearest_gene.R" class="external-link"><code>R/get_nearest_gene.R</code></a></small>
      <div class="d-none name"><code>get_nearest_gene.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Use distance variant position to get the nearest gene. Uses snpsettest function `map_snp_to_gene` to identify genes from GENCODE databases (https://github.com/HimesGroup/snpsettest).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_nearest_gene</span><span class="op">(</span></span>
<span>  <span class="va">variants</span>,</span>
<span>  detect_headers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  build <span class="op">=</span> <span class="fl">37</span>,</span>
<span>  n_bases <span class="op">=</span> <span class="fl">1e+05</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>variants</dt>
<dd><p>A data.frame. Contains the variants (e.g., in summary statistics).</p></dd>


<dt>detect_headers</dt>
<dd><p>Logical. Default=TRUE. Search input headers to see if BOLT-LMM, SAIGE, or REGENIE input (user therefore doesn't need to provide).</p></dd>


<dt>snp_col</dt>
<dd><p>A string. Default="SNP". The RSID/variantID column name.</p></dd>


<dt>chr_col</dt>
<dd><p>A string. Default="CHR". The chromosome column name.</p></dd>


<dt>pos_col</dt>
<dd><p>A string. Default="BP". The base pair/position column name.</p></dd>


<dt>build</dt>
<dd><p>An integer. Default=37. Genome build to use (can only be 37 or 38).</p></dd>


<dt>n_bases</dt>
<dd><p>An interger. Default=1e5. The max distance in base-pairs between a variant and a gene to annotate</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns a data frame of variant IDs mapped to genes (with distance).</p>


<p>If `dist` is positive, the variant is intergenic, and this is the distance to the closest gene. If `dist` is negative, the variant is within a gene, and this is the distance to the start of the gene. If `dist` is NA, the variant is not within `n_bases` of a gene in GENCODE.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Luke Pilling</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu"><a href="get_loci.html">get_loci</a></span><span class="op">(</span><span class="va">gwas_example</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Locus size (bases) = 5e+05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P-value threshold = 5e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> N variants = 319732</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> N variants p&lt;threshold = 4132</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> N loci = 15</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gwas_loci_genes</span> <span class="op">=</span> <span class="fu">get_nearest_gene</span><span class="op">(</span><span class="va">gwas_loci</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Using human genome build 37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting nearest gene for 4131 unique variants</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> (Removed 1 duplicated or missing variant IDs/positions)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci_genes</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           SNP CHR        BP   A1 A2   MAF       BETA         SE        P locus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  rs12046439   1 107536799    T  C 0.248 0.00997159 0.00170546 5.01e-09     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 rs143849791   1 107537916 CATG  C 0.325 0.01283200 0.00164361 5.85e-15     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 rs113329442   1 107539252    A  G 0.330 0.01109240 0.00149706 1.27e-13     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   rs3861909   1 107544176    G  A 0.327 0.01187220 0.00150837 3.52e-15     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  rs17496332   1 107546375    A  G 0.331 0.01110260 0.00148844 8.70e-14     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   rs2878349   1 107549245    G  A 0.327 0.01182020 0.00149200 2.33e-15     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    lead  gene  dist</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 FALSE PRMT6 62468</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 FALSE PRMT6 61351</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 FALSE PRMT6 60015</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 FALSE PRMT6 55091</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 FALSE PRMT6 52892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 FALSE PRMT6 50022</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci_genes</span><span class="op">[</span> <span class="va">gwas_loci_genes</span><span class="op">$</span><span class="va">lead</span><span class="op">==</span><span class="cn">TRUE</span> , <span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              SNP CHR        BP A1 A2     MAF        BETA         SE         P</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13   rs111232683   1 107566149  G  C 0.34300  0.01352040 0.00161401  5.43e-17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55   rs114254196   1 108635400  C  T 0.00848 -0.04481140 0.00818473  4.38e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92   rs115292790   1 109310728  G  A 0.01360 -0.05639270 0.00608890  2.01e-20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 672   rs12740374   1 109817590  G  T 0.21900 -0.14822800 0.00166391 4.73e-305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1542 rs140266316   1 110326545  G  A 0.01630 -0.05770880 0.00597770  4.73e-22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1564    rs657801   1 111736389  T  C 0.31500  0.00905412 0.00150713  1.88e-09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      locus lead     gene   dist</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13       1 TRUE    PRMT6  33118</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55       2 TRUE SLC25A24  41258</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92       3 TRUE   STXBP3 -21432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 672      4 TRUE   CELSR2 -24949</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1542     5 TRUE    GSTM5   8495</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1564     6 TRUE  DENND2D  -6593</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://lukepilling.github.io" class="external-link">Luke Pilling</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

