<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Use distance from lead significant SNP to estimate independet loci in GWAS summary stats. Uses -log10(p) derived from BETA/SE so does not need P as input."><title>Get loci from GWAS file — get_loci • gwasRtools</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get loci from GWAS file — get_loci"><meta property="og:description" content="Use distance from lead significant SNP to estimate independet loci in GWAS summary stats. Uses -log10(p) derived from BETA/SE so does not need P as input."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gwasRtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.3.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lukepilling/gwasRtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Get loci from GWAS file</h1>
      <small class="dont-index">Source: <a href="https://github.com/lukepilling/gwasRtools/blob/HEAD/R/get_loci.R" class="external-link"><code>R/get_loci.R</code></a></small>
      <div class="d-none name"><code>get_loci.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Use distance from lead significant SNP to estimate independet loci in GWAS summary stats. Uses -log10(p) derived from BETA/SE so does not need P as input.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_loci</span><span class="op">(</span></span>
<span>  <span class="va">gwas</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  maf_col <span class="op">=</span> <span class="st">"MAF"</span>,</span>
<span>  beta_col <span class="op">=</span> <span class="st">"BETA"</span>,</span>
<span>  se_col <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>  stat_col <span class="op">=</span> <span class="st">"NA"</span>,</span>
<span>  p_col <span class="op">=</span> <span class="st">"NA"</span>,</span>
<span>  neglog10p_col <span class="op">=</span> <span class="st">"NA"</span>,</span>
<span>  use_pvalue <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  detect_headers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_bases <span class="op">=</span> <span class="fl">5e+05</span>,</span>
<span>  p_threshold <span class="op">=</span> <span class="fl">5e-08</span>,</span>
<span>  exclude_hla <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  hla_pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2.5e+07</span>, <span class="fl">3.4e+07</span><span class="op">)</span>,</span>
<span>  get_ld_indep <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ld_pruning_r2 <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  ld_clump_local <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ld_plink_bin <span class="op">=</span> <span class="st">"plink"</span>,</span>
<span>  ld_bfile <span class="op">=</span></span>
<span>    <span class="st">"/indy/ukbiobank/data_14631/genetics/imputed_500k/5k_eur/ukb_imp_v3.qc_sub.5k_eur"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>gwas</dt>
<dd><p>A data.frame. Contains the GWAS summary statistics.</p></dd>


<dt>snp_col</dt>
<dd><p>A string. Default="SNP". The RSID/variantID column name.</p></dd>


<dt>chr_col</dt>
<dd><p>A string. Default="CHR". The chromosome column name.</p></dd>


<dt>pos_col</dt>
<dd><p>A string. Default="BP". The base pair/position column name.</p></dd>


<dt>maf_col</dt>
<dd><p>A string. Default="MAF". The MAF/minor-allele-frequency column name.</p></dd>


<dt>beta_col</dt>
<dd><p>A string. Default="BETA". The BETA column name.</p></dd>


<dt>se_col</dt>
<dd><p>A string. Default="SE". The SE column name.</p></dd>


<dt>stat_col</dt>
<dd><p>A string. Default="NA". The test statistic column name. (Only required if not providing beta+se, or neglog10p)</p></dd>


<dt>p_col</dt>
<dd><p>A string. Default="NA". The p-value column name. (Only required if `use_pvalue==TRUE` i.e., using the provided p-value, e.g., P_BOLT_LMM)</p></dd>


<dt>neglog10p_col</dt>
<dd><p>A string. Default="NA". The -log10 p-value column name. (Only required if not providing beta+se, or stat)</p></dd>


<dt>use_pvalue</dt>
<dd><p>Logical. Default=FALSE. Use the provided p-value (in `p_col`) rather than computing from the test statistic? Useful for BOLT-LMM output</p></dd>


<dt>detect_headers</dt>
<dd><p>Logical. Default=TRUE. Search input headers to see if BOLT-LMM, SAIGE, or REGENIE input (user therefore doesn't need to provide). If BOLT-LMM then automatically use p-value instead of SE.</p></dd>


<dt>n_bases</dt>
<dd><p>An interger. Default=5e5. The distance between two significant loci, beyond which they are defined as in separate loci.</p></dd>


<dt>p_threshold</dt>
<dd><p>A number. Default=5e-8. P-value threshold for statistical significance</p></dd>


<dt>exclude_hla</dt>
<dd><p>Logical. Default=FALSE. Treat HLA as one continuous locus</p></dd>


<dt>hla_pos</dt>
<dd><p>A numeric vector of length 2. Default=c(25e6, 34e6). The HLA region on chromosome 6 to treat as one continuous locus if `exclude_hla==TRUE`</p></dd>


<dt>get_ld_indep</dt>
<dd><p>Logical. Default=FALSE. Use Plink LD clumping to identify independent SNPs - see ieugwasr::ld_clump() docs</p></dd>


<dt>ld_pruning_r2</dt>
<dd><p>Numeric. Default=0.01. Pruning threshold for LD.</p></dd>


<dt>ld_clump_local</dt>
<dd><p>Logical. Default=TRUE. If clumping using local installation (rather than IEU API) - see ieugwasr::ld_clump() docs</p></dd>


<dt>ld_plink_bin</dt>
<dd><p>A string. Default="plink". Path to Plink v1.90 binary</p></dd>


<dt>ld_bfile</dt>
<dd><p>A string. Default is to 5,000 random unrelated UK Biobank Europeans on my server :) needs a path to appropriate BIM/BED reference panel files on your server</p></dd>


<dt>verbose</dt>
<dd><p>Logical. Default=FALSE. Be verbose</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns a data frame of significant SNPs with locus annotation</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Luke Pilling</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gwas_loci</span> <span class="op">=</span> <span class="fu">get_loci</span><span class="op">(</span><span class="va">gwas_example</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Locus size (bases) = 5e+05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P-value threshold = 5e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in match(x, table, nomatch = 0L):</span> 'match' requires vector arguments</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(expr, envir, enclos):</span> object 'gwas_loci' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gwas_loci</span><span class="op">[</span> <span class="va">gwas_loci</span><span class="op">$</span><span class="va">lead</span><span class="op">==</span><span class="cn">TRUE</span> , <span class="op">]</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(expr, envir, enclos):</span> object 'gwas_loci' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://lukepilling.github.io" class="external-link">Luke Pilling</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

